src/System Application/App/AppSource Gallery/src/AppSourceProductManager.Codeunit.al

```diff
@@ -0,0 +1,604 @@
+        exit(Language.GetFormatRegionOrDefault(FormatRegion));
+    end;
+
+    procedure RestClient_GetAsJSon(var RestClient: Codeunit "Rest Client"; RequestUri: Text): JsonToken
+    begin
+        exit(RestClient.GetAsJSon(RequestUri));
+    end;
+
+    procedure UserSettings_GetUserSettings(UserSecurityId: Guid; var TempUserSettingsRecord: record "User Settings" temporary)
+    var
+        UserSettings: Codeunit "User Settings";
+    begin
+        UserSettings.GetUserSettings(Database.UserSecurityID(), TempUserSettingsRecord);
+    end;
+
+    #endregion
+    procedure SetDependencies(SpecificDependencies: Interface "IAppSource Product Manager Dependencies")
+    begin
+        Dependencies := SpecificDependencies;
+        IsDependenciesSet := true;
+    end;
+
+    #region Product helpers 
+    /// <summary>
+    /// Opens Microsoft AppSource web page.
+    /// </summary>
+    procedure OpenAppSource()
+    begin
+        Init();
+        Hyperlink(StrSubstNo(AppSourceUriLbl, Dependencies.Language_GetFormatRegionOrDefault('en-us')));
+    end;
+
+
+    /// <summary>
+    /// Opens the AppSource product page in Microsoft AppSource, for the specified unique product ID.
+    /// </summary>
+    /// <param name="UniqueProductIDValue">The Unique Product ID of the product to show in MicrosoftAppSource</param>
+    procedure OpenInAppSource(UniqueProductIDValue: Text)
+    var
+        Language: Codeunit Language;
+    begin
+        Hyperlink(StrSubstNo(AppSourceListingUriLbl, Language.GetCultureName(GetCurrentUserLanguageID()), UniqueProductIDValue));
+    end;
+
+    /// <summary>
+    /// Opens the AppSource product details page for the specified unique product ID.
+    /// </summary>
+    /// <param name="UniqueProductIDValue"></param>
+    internal procedure OpenProductDetailsPage(UniqueProductIDValue: Text)
+    var
+        ProductDetailsPage: Page "AppSource Product Details";
+        ProductObject: JsonObject;
+    begin
+        ProductObject := GetProductDetails(UniqueProductIDValue);
+        ProductDetailsPage.SetProduct(ProductObject);
+        ProductDetailsPage.RunModal();
+    end;
+
+    /// <summary>
+    /// Extracts the AppID from the Unique Product ID.
+    /// </summary>
+    /// <param name="UniqueProductIDValue">The Unique Product ID of the product as defined in MicrosoftAppSource</param>
+    /// <returns>AppID found in the Product ID</returns>
+    /// <remarks>The AppSource unique product ID is specific to AppSource and combines different features while always ending with PAPID. and extension app id. Example: PUBID.mdcc1667400477212|AID.bc_converttemp_sample|PAPPID.9d314b3e-ffd3-41fd-8755-7744a6a790df</remarks>
+    internal procedure ExtractAppIDFromUniqueProductID(UniqueProductIDValue: Text): Text[36]
+    var
+        AppIDPos: Integer;
+    begin
+        AppIDPos := StrPos(UniqueProductIDValue, 'PAPPID.');
+        if (AppIDPos > 0) then
+            exit(CopyStr(UniqueProductIDValue, AppIDPos + 7, 36));
```

Comment:
Since the AppId is a guid, might be nice to return a guid instead of a string. That also ensures we read this correctly.

Reply:
Next PR